### PR TITLE
build and upload images when publishing

### PR DESCRIPTION
**Pull request by zach593**
Signed-off-by: zach593 <zach_li@outlook.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

build and upload images with static tags when release published

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:

split from https://github.com/karmada-io/karmada/pull/1358

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -44,3 +44,27 @@ jobs:
     - uses: actions/checkout@master
     - name: Update new version in krew-index
       uses: rajatjindal/krew-release-bot@v0.0.40
+  publish-image:
+    name: publish images
```
Comment (RainbowMango) [inline .github/workflows/release.yml:48]:
```suggestion
  release-image:
    name: release image
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -44,3 +44,27 @@ jobs:
...
+    # prevent job running from forked repository, otherwise
+    # 1. running on the forked repository would fail as missing necessary secret.
+    # 2. running on the forked repository would use unnecessary GitHub Action time.
+    if: ${{ github.repository == 'karmada-io/karmada' }}
+    runs-on: ubuntu-18.04
+    steps:
+    - name: checkout code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (RainbowMango) [inline .github/workflows/release.yml:58]: Just a question, it defaults to check out the code from the master branch, right?
What if we cut a release from the release branch? like cut `v1.0.2` from branch `release-1.0`?

**Review by RainbowMango — COMMENTED** [review]
I suggest moving this job to a standalone workflow file, e.g. `release-image.yml`.

**Comment by zach593** [comment]
quote from https://github.com/actions/checkout#usage
```
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
    ref: ''
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (zach593) [reply .github/workflows/release.yml:48]: should it always be "publish a release"?

Comment (RainbowMango) [reply .github/workflows/release.yml:48]: ?
This workflow takes the responsibility of releasing the images when we cut a release.

**Comment by RainbowMango** [comment]
> quote from https://github.com/actions/checkout#usage
> 
> ```
>     # The branch, tag or SHA to checkout. When checking out the repository that
>     # triggered a workflow, this defaults to the reference or SHA for that event.
>     # Otherwise, uses the default branch.
>     ref: ''
> ```

Good, that's fine.

File: .github/workflows/release.yml (hunk 1 above)
Comment (zach593) [reply .github/workflows/release.yml:48]: forget it, doesn't matter, just focused on "release" should be a noun or a verb

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**